### PR TITLE
[Robot] RD2 Testcase to assert no duplicate opportunities - usecase #2

### PR DESCRIPTION
**Pull request by rjanjanam**
https://foundation.lightning.force.com/lightning/r/agf__ADM_Work__c/a2x1E000007hbsgQAA/view
The above ticket has 4 use-cases. This PR covers 2nd and 3rd use-cases
- Create a Recurring Donation from backend API and ensure there is right number of opportunities created. Update the opportunity stage to closed-won and then edit the RD2 day of the month to a different date.
- Validate that there is a new open opportunity and ensure the closing dates reflect as expected on both the opportunities.

# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: robot/Cumulus/tests/browser/recurring_donations/zedit_day_of_month_verify_next_opportunity.robot [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+    #Validate that the number of opportunities now show as 2 . Verify the details on the respective opportunities
+    Validate Related Record Count           Opportunities                                                  2
+    @{opportunity} =                        API Query Opportunity For Recurring Donation                  ${data}[contact_rd][Id]
+    Validate Opportunity Details            ${opportunity}[0][Id]        Closed Won                       ${CURRENT_DATE}
+    Validate Opportunity Details            ${opportunity}[1][Id]        Pledged                          ${next_payment_date}
+
+
+
+
```
Comment (Safiyab) [inline robot/Cumulus/tests/browser/recurring_donations/zedit_day_of_month_verify_next_opportunity.robot:95]: Can you remove this blank lines ? Lint will throw a `TrailingBlankLines` error. Also, run the robot_lint task and remove errors, if any.

Comment (rjanjanam) [reply robot/Cumulus/tests/browser/recurring_donations/zedit_day_of_month_verify_next_opportunity.robot]: Thanks, will run the lint task and make the required changes

File: robot/Cumulus/resources/RecurringDonationsPageObject.py [hunk 2]
```diff
@@ -182,6 +182,16 @@ def validate_field_values_under_section(self, section=None, **kwargs):
                     label, "contains", value, section
                 )
 
+    def get_next_payment_date_number(self, num):
+        """Returns the next payment date from the payment schedule"""
```
Comment (skristem) [inline robot/Cumulus/resources/RecurringDonationsPageObject.py:186]: can you please clarify on what is the num variable here

Comment (rjanjanam) [reply robot/Cumulus/resources/RecurringDonationsPageObject.py]: Updated the documentation to be more clearer with example

**Pull request merged**